src/BitStringAddresses/fermifs.jl

```diff
@@ -0,0 +1,213 @@
+struct FermiFS{N,M,S<:BitString{M}} <: AbstractFockAddress
+    bs::S
+
+    FermiFS{N,M,S}(bs::S) where {N,M,S<:BitString{M}} = new{N,M,S}(bs)
+end
+
+function FermiFS{N,M,S}(onr::Union{SVector{M},NTuple{M}}) where {N,M,C,T,S<:BitString{M,C,T}}
+    @boundscheck sum(onr) == N && all(in((0, 1)), onr) || error("invalid ONR")
+    result = zero(SVector{C,T})
+    for orbital in 1:M
+        iszero(onr[orbital]) && continue
+        minus_j, offset = fldmod(orbital - 1, 64)
+        j = C - minus_j
+        new = result[j] | T(1) << T(offset)
+        result = setindex(result, new, j)
+    end
+    return FermiFS{N,M,S}(S(SVector(result)))
+end
+
+function FermiFS{N,M}(onr::Union{AbstractVector,Tuple}) where {N,M}
+    S = typeof(BitString{M}(0))
+    return FermiFS{N,M,S}(SVector{M}(onr))
+end
+function FermiFS{N}(onr::Union{SVector{M},Tuple{M}}) where {N,M}
+    return FermiFS{N,M}(onr)
+end
+function FermiFS(onr::Union{AbstractVector,Tuple})
+    M = length(onr)
+    N = sum(onr)
+    return FermiFS{N,M}(onr)
+end
+
+function Base.show(io::IO, f::FermiFS{N,M}) where {N,M}
+    print(io, "FermiFS{$N,$M}(", tuple(onr(f)...), ")")
+end
+
+Base.isless(a::FermiFS, b::FermiFS) = isless(a.bs, b.bs)
+Base.hash(a::FermiFS,  h::UInt) = hash(a.bs, h)
+Base.:(==)(a::FermiFS, b::FermiFS) = a.bs == b.bs
+num_particles(::Type{FermiFS{N,M,S}}) where {N,M,S} = N
+num_modes(::Type{FermiFS{N,M,S}}) where {N,M,S} = M
+num_components(::Type{<:FermiFS}) = 1
+
+function near_uniform(::Type{FermiFS{N,M}}) where {N,M}
+    return FermiFS([fill(1, N); fill(0, M - N)])
+end
+
+onr(a::FermiFS) = SVector(m_onr(a))
+
+struct FermiOccupiedOrbitals{N,S}
+    bs::S
+end
+
+"""
+    occupied_orbitals(f::FermiFS)
+
+Iterate over occupied orbitals in `FermiFS` address. Iterates values of type `Int`.
+
+# Example
+
+```jldoctest
+julia> f = FermiFS((1,1,0,1,0,0,1))
+julia> for i in occupied_orbitals(f)
+    @show i
+end
+i = 1
+i = 2
+i = 4
+i = 7
+
+"""
+occupied_orbitals(a::FermiFS{N,<:Any,S}) where {N,S} = FermiOccupiedOrbitals{N,S}(a.bs)
+
```

Comment:
```suggestion

Base.length(o::FermiOccupiedOrbitals{N}) where {N} = N
```
needed for iterator interface, e.g. `collect`.

If we make it an exported function and part of the API then we should have it. But maybe this will be more of an internal function? In that case we would not need it, I guess. 

The method for `BoseFS` should get the same treatment.

And maybe a test via `collect`?